src/http-api/error-handler.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+const Hoek = require('hoek')
+
+module.exports = (server) => {
+  server.ext('onRequest', (request, reply) => {
+    request.handleError = handleError
+    reply.continue()
+  })
+
+  server.ext('onPreResponse', (request, reply) => {
+    const res = request.response
+    const req = request.raw.req
+
+    let statusCode = 200
+    let msg = 'Sorry, something went wrong, please retrace your steps.'
+
+    if (res.isBoom) {
+      statusCode = res.output.payload.statusCode
+
+      if (res.message && res.isDeveloperError) {
+        // we caught it!
```

Comment:
Need to clean up this file and add some better tests for it, this is mostly ported from another module atm, which is why the comments are a bit odd